### PR TITLE
test: e2e replay test using Maestro (js)

### PR DESCRIPTION
**Pull request by vaind**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description

Adds integration tests for Replay by updating e2e tests to:

- attach replays on errors (change the init config)
- check that a replay is created on sentry.io

During development, I've found out it's possible to launch javascript from maestro directly, which reduces development iteration time (make a change in tests and see the result) **from about 9 minutes to 20 seconds**, because we don't actually need to rebuild the whole app when making a change to test assertions.

Note: android doesn't seem to capture replays in CI so it's disabled.
Note 2: metrics failure isn't related IMO

## :bulb: Motivation and Context

Replaces #3993 

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: dev-packages/e2e-tests/maestro/captureUnhandledPromiseRejection.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 appId: ${APP_ID}
+jsEngine: graaljs
 ---
 - runFlow: utils/launchTestAppClear.yml
-- tapOn: "Unhandled Promise Rejection"
+- tapOn: 'Unhandled Promise Rejection'
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/captureUnhandledPromiseRejection.yml:5]:
````suggestion
- tapOn: "Unhandled Promise Rejection"
````

File: dev-packages/e2e-tests/maestro/captureException.yml [hunk 2]
```diff
@@ -1,5 +1,10 @@
 appId: ${APP_ID}
+jsEngine: graaljs
 ---
 - runFlow: utils/launchTestAppClear.yml
-- tapOn: "Capture Exception"
+- tapOn: 'Capture Exception'
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/captureException.yml:5]:
````suggestion
- tapOn: "Capture Exception"
````

File: dev-packages/e2e-tests/maestro/utils/launchTestAppClear.yml [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+- killApp
+
 - launchApp:
     clearState: true
     arguments:
       sentryAuthToken: ${SENTRY_AUTH_TOKEN}
 
 - extendedWaitUntil:
-    visible: "E2E Tests Ready"
+    visible: 'E2E Tests Ready'
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/utils/launchTestAppClear.yml:14]:
````suggestion
    visible: "E2E Tests Ready"
````

File: dev-packages/e2e-tests/maestro/utils/assertEventIdVisible.yml [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
 ---
 - extendedWaitUntil:
     visible:
-      id: "eventId"
+      id: 'eventId'
     timeout: 600_000 # 10 minutes
 
 - copyTextFrom:
-    id: "eventId"
+    id: 'eventId'
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/utils/assertEventIdVisible.yml:10]:
````suggestion
    id: "eventId"
````

File: dev-packages/e2e-tests/maestro/utils/assertEventIdVisible.yml [hunk 5]
```diff
@@ -1,10 +1,20 @@
 appId: ${APP_ID}
+jsEngine: graaljs
 ---
 - extendedWaitUntil:
     visible:
-      id: "eventId"
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/utils/assertEventIdVisible.yml:5]:
````suggestion
    id: "eventId"
````

File: dev-packages/e2e-tests/maestro/utils/assertEventIdVisible.yml [hunk 6]
```diff
@@ -1,10 +1,20 @@
...
+      id: 'eventId'
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/utils/assertEventIdVisible.yml:6]:
````suggestion
      id: "eventId"
````

File: dev-packages/e2e-tests/maestro/captureException.yml [hunk 7]
```diff
@@ -1,5 +1,10 @@
...
 - runFlow: utils/assertEventIdVisible.yml
+- runFlow:
+    file: utils/assertReplay.yml
+    when:
+      platform: iOS
```
Comment (krystofwoldrich) [inline dev-packages/e2e-tests/maestro/captureException.yml:10]: I would create standalone replay on error test and enable replay thru launch args. 

To separate capturing errors and capturing replay on error.

File: dev-packages/e2e-tests/maestro/utils/sentryApi.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  for (const [key, value] of Object.entries(data)) {
+    console.log(`Setting output.${key} = '${value}'`);
+    output[key] = value;
+  }
+}
+
+// Note: "fetch", "id", "eventId", etc. are script inputs, see for example assertEventIdIVisible.yml
+switch (fetch) {
+  case 'event': {
+    const data = json(fetchFromSentry(`${baseUrl}/events/${id}/json/`));
```
[the remark this answers is not in the document]
Comment (vaind) [reply dev-packages/e2e-tests/maestro/utils/sentryApi.js:60]: maestro

File: dev-packages/e2e-tests/maestro/captureException.yml (hunk 7 above)
Comment (vaind) [reply dev-packages/e2e-tests/maestro/captureException.yml]: done

File: dev-packages/e2e-tests/maestro/utils/sentryApi.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply dev-packages/e2e-tests/maestro/utils/sentryApi.js:60]: Thanks, that what I was looking for.

File: dev-packages/e2e-tests/maestro/captureException.yml (hunk 7 above)
Comment (vaind) [reply dev-packages/e2e-tests/maestro/captureException.yml]: bd6dc14fd31058432bd8ec794d8b0e76ec0a7dc6

**Review by krystofwoldrich — APPROVED** [review]
Thank you for exploring the graaljs,

this is good improvement to the maestro mobile tests.

🚀

**Pull request merged**